vars/spinInfra.groovy

```diff
@@ -8,20 +8,48 @@ import uk.gov.hmcts.pipeline.AKSSubscriptions
 import uk.gov.hmcts.contino.RepositoryUrl
 
 def call(productName, environment, tfPlanOnly, subscription) {
-  call(productName, null, environment, tfPlanOnly, subscription)
+  call(
+    product: params.product,
+    environment: params.environment,
+    tfPlanOnly: params.planOnly,
+    subscription: params.subscription
+  )
 }
 
 def call(product, component, environment, tfPlanOnly, subscription) {
-  call(product, component, environment, tfPlanOnly, subscription, "")
+  call(
+    product: params.product,
+    component: params.component ?: null,
+    environment: params.environment,
+    tfPlanOnly: params.planOnly,
+    subscription: params.subscription
+  )
 }
 
 def call(product, component, environment, tfPlanOnly, subscription, deploymentTarget) {
-  def branch = new ProjectBranch(env.BRANCH_NAME)
+  call(
+    product: params.product,
+    component: params.component ?: null,
+    environment: params.environment,
+    tfPlanOnly: params.planOnly,
+    subscription: params.subscription,
+    deploymentTarget: params.deploymentTarget ?: null
+  )
+}
 
-  def deploymentNamespace = branch.deploymentNamespace()
+def call(params) {
+  def product = params.product
+  def component = params.component ?: null
+  def expires = params.expires
+  def environment = params.environment
+  def tfPlanOnly = params.planOnly
+  def subscription = params.subscription 
   def productName = component ? "$product-$component" : product
+  def branch = new ProjectBranch(env.BRANCH_NAME)
+  def deploymentNamespace = branch.deploymentNamespace()
   def changeUrl = ""
   def environmentDeploymentTarget = "$environment"
+  def deploymentTarget
```

Comment:
Is there a benefit to using that pattern? Looks to me like everything is being set twice in organisations-beta.groovy e.g. https://github.com/hmcts/cnp-jenkins-config/blob/447c061474c4e20cd2eb3770ab2e151899471785/jobdsl/organisations-beta.groovy#L51 and https://github.com/hmcts/cnp-jenkins-config/blob/447c061474c4e20cd2eb3770ab2e151899471785/jobdsl/organisations-beta.groovy#L63

Reply:
Yes, this is everything (8 values):
https://github.com/hmcts/cnp-jenkins-config/blob/447c061474c4e20cd2eb3770ab2e151899471785/jobdsl/organisations-beta.groovy#L26-L33

 and the second pass is for values that may be overridden based on conditions but means you specify 4 values instead of 8.